client/operations.go

```diff
@@ -148,6 +177,10 @@ func (op *operation) WaitContext(ctx context.Context) error {
 // It adds handlers to process events, monitors the listener for completion or errors,
 // and triggers a manual refresh of the operation's state to prevent race conditions.
 func (op *operation) setupListener() error {
+	if op.skipListener {
```

Comment:
Can we return an error here, so that calling setupListener with op.skipListener is invalid?

Reply:
Done.